### PR TITLE
Correcting multiple bugs

### LINKED ISSUE #206: het_goldfeldquandt
Hello,

There are 2 problems with het_goldfeldquandt function. I compare verticapy function with statsmodels.stats.diagnostic function, and I use this documentation : https://fr.wikipedia.org/wiki/Test_de_Goldfeld_et_Quandt

In verticapy, with this code : 

```
from verticapy.stats import het_goldfeldquandt
het_goldfeldquandt(amazon_unique_copy, 
                   y = "number_sum", 
                   X = ["random"],
                   idx = 0, 
                   split = 0.5)
```

I have the result : 
image

In  python with this code : 

```
import statsmodels.stats.diagnostic
df_amazon_het_goldfeldquandt = amazon_unique.to_pandas()
df_amazon_het_goldfeldquandt = df_amazon_het_goldfeldquandt.sort_values("date")
df_amazon_het_goldfeldquandt.index = [i for i in range(len(df_amazon_het_goldfeldquandt))]
statsmodels.stats.diagnostic.het_goldfeldquandt(df_amazon_het_goldfeldquandt["number_sum"], df_amazon_het_goldfeldquandt[["random"]], idx = 0,split = 0.5, alternative = "increasing")

```

I have the result : 
image

The results are different. I code this function with : 

```
amazon_copy = amazon_unique.copy()
split_value = amazon_copy[["random"][0]].quantile(0.5)
amazon_copy1 = amazon_copy.search(amazon_copy[["random"][0]] < split_value)
amazon_copy2 = amazon_copy.search(amazon_copy[["random"][0]] > split_value)

drop(schema + ".amazon_lr_test",cur,method = "model")
model1 = LinearRegression(schema + ".amazon_lr_test")
model1.drop()
model1.fit(amazon_copy1, X = ["random"], y = "number_sum")
model1.predict(amazon_copy1, name = "number_pred1")
amazon_copy1["eps1"] = amazon_copy1["number_sum"] - amazon_copy1["number_pred1"]

drop(schema + ".amazon_lr_test",cur,method = "model")
model2 = LinearRegression(schema + ".amazon_lr_test")
model2.drop()
model2.fit(amazon_copy2, X = ["random"], y = "number_sum")
model2.predict(amazon_copy2, name = "number_pred2")
amazon_copy2["eps2"] = amazon_copy2["number_sum"] - amazon_copy2["number_pred2"]
```

To finish, I calcul the ratio in 4 different ways : 

image

Verticapy result is le thirds way, python result is the first way. In wikipédia and in python, they use MSE, not VAR. "The test statistic used is the ratio of the mean square residual errors for the regressions on the two subsets" in english version of wikipédia. 

the second problem is that the ratios are reversed. In python is high values mse/ low values mse, in wikipédia we are f = SCR2 / SCR1 with scr2 high value mse et scr1 low values mse. 

Regards,

Alexandre

### LINKED ISSUE #217: aic_bic
hello,

for this calcul, you keep all lines, but we must drop the line where the prediction is null. In your code, "n" must be : number of lines without null in prediction. 

image

regards

Alexandre

### LINKED ISSUE #227: NearestCentroid method get_attr
Hello,

For NearestCentroid function, I have an error for get_attr method. 

With this code : 

```
from verticapy.learn.neighbors import NearestCentroid

drop(name = schema +".model_titanic_test", cursor = cur, method = "model")
model = NearestCentroid(name = schema +".model_titanic_test", cursor = cur, p = 20)
model.fit(titanic,["age","fare"], "survived")
model.get_attr()
```

I have the following error : 

image

why?

Regards,

Alexandre

### LINKED ISSUE #228: scatter, public schema
Hello,

method scatter create temporary view in public schema, but we don't have access to this schema. 

the error is : 
image

can you add parameter schema for this method to write temporary object? 

Regards,

Alexandre

### PR DESCRIPTION
**Pull request by oualib**
This PR will solve multiple issues:
public schema problem: 
Resolves #228 #229 #230 #231 #232 #233 #234 #235 #236 #237 #238 
We created a global variable which will represent the temporary schema used to write temp object. This parameter can be modified via the set_option function.
some models had problems with get_attr method: 
Resolves #227, #212
some metrics did not handle missing values: 
Resolves #217, #218, #219, #220
some statistical tests were incorrect: 
Resolves #206, #207, #208, #194

### CODE REVIEW

File: verticapy/learn/metrics.py [hunk 1]
```diff
@@ -169,11 +169,11 @@ def anova_table(
         ]
     )
     cursor, conn, input_relation = check_cursor(cursor, input_relation)
-    query = "SELECT COUNT(*), AVG({}) FROM {}".format(y_true, input_relation)
+    query = "SELECT COUNT(*), AVG({}) FROM {} WHERE {} IS NOT NULL AND {} IS NOT NULL;".format(y_true, input_relation, y_true, y_score,)
```
Comment (afard) [inline verticapy/learn/metrics.py:172]: Do we need the part related to `y_score` in this query?

Comment (afard) [reply verticapy/learn/metrics.py:172]: Looking at the usage of `avg` in the next query, I guess we need.

**Review by ansleis — APPROVED** [review]
Looks good, made some minor copyedits. Let me know if I missed anything.

File: verticapy/tests/stats/test_stats.py [hunk 2]
```diff
@@ -125,13 +125,23 @@ def test_het_breuschpagan(self, amazon_vd):
...
-        result = amazon_vd.groupby(["date"], ["AVG(number) AS number"])
-        result["lag_number"] = "LAG(number) OVER (ORDER BY date)"
-        result = st.het_goldfeldquandt(result, y="number", X=["lag_number"])
+        vdf = amazon_vd.groupby(["date"], ["AVG(number) AS number"])
+        vdf["lag_number"] = "LAG(number) OVER (ORDER BY date)"
+        result = st.het_goldfeldquandt(vdf, y="number", X=["lag_number"])
         assert result["value"] == [
-            pytest.approx(0.0331426182368922),
+            pytest.approx(30.17263128858259),
+            pytest.approx(1.3988910574921388e-55),
```
Comment (afard) [inline verticapy/tests/stats/test_stats.py:133]: This number is very small. I wonder if it would make the test unstable.

**Review by afard — APPROVED** [review]

File: verticapy/tests/stats/test_stats.py (hunk 2 above)
Comment (oualib) [reply verticapy/tests/stats/test_stats.py:133]: it will but we just need a value near to 0.0

**Pull request merged**